examples/simple-grep/README.md

```diff
@@ -0,0 +1,129 @@
+There is a basic example usage of Iris and a tutorial-like explanation:
+
+First of all, let's create a newtype wrapper for the main monad of Iris:
+
+```haskell
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+
+module Main (main) where
+
+import Prelude hiding (readFile)
+import Control.Monad.IO.Class (MonadIO (..))
+import Control.Monad.Reader (MonadReader)
+
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.ByteString as BS
+import qualified Data.Text.Lazy as T
+import qualified Data.Text.Lazy.Encoding as TLE
+import qualified Options.Applicative as Opt
+
+import qualified Colourista
+import qualified Iris
+import qualified Paths_iris as Autogen
```

Comment:
Usually, this part is called "Preamble" in LHS tutorials. So it's beneficial to move all language extensions and imports into a separate `## Preamble` section and separate code block.

I would go even further and probably moved `Iris` import and `Paths_iris` import into separate sections with extra comments.